_posts/2018-11-29-async-update.md

```diff
@@ -0,0 +1,201 @@
+
+[boats-syntax]: https://boats.gitlab.io/blog/post/await-syntax/
+[existential types]: https://github.com/rust-lang/rfcs/pull/2071
+
+# The Supporting APIs
+
+Like many other language features, async/await also requires some support in the
+standard library: shipping the `Future` trait (and associated machinery) in `std`.
+That work has been a major thrust this year, and is nearing completion.
+
+## Core futures APIs
+
+There's currently an [open RFC][futures-rfc] proposing stabilization of the futures
+APIs, and includes a [fairly detailed writeup][futures-history] of the history of
+those APIs. The pull request contains a checklist of current blockers; the most
+signifcant one at the moment is finalizing the `Waker` APIs.
+
+[futures-history]: https://github.com/aturon/rfcs/blob/future/text/0000-futures.md#historical-context
+[futures-rfc]: https://github.com/rust-lang/rfcs/pull/2592
+
+## The `Pin` API
+
+One of the underlying mechanisms supporting the futures API is the `Pin` type,
+which is also how we enable borrowing in `async` blocks. This API, too, has seen
+significant iteration over the course of the year. @withoutboats's [blog post][pin-final-api]
+from a few months ago covers the final design, which has also been
+[proposed for stabilization][pin-stabilization]. The only remaining sticking point
+is around type and trait naming.
+
+[pin-final-api]: https://boats.gitlab.io/blog/post/rethinking-pin/
+[pin-stabilization]: https://github.com/rust-lang/rust/issues/55766
+
+## Compatibility with futures 0.1
+
+The design of the futures API had to change in breaking ways in order to support
+async/await. However, there's a large existing ecosystem of code that uses the earlier
+futures 0.1 API. Luckily, we're able to provide a rather ergonomic compatibility layer
+that makes it possible to move between the two APIs easily, and hence support
+incremental migration. A [recent blog post from @jsdw][futures-compat] does an excellent
+job of laying out how this compatibility story works.
+
+[futures-compat]: https://jsdw.me/posts/rust-asyncawait-preview/
+
+# Some new crates
+
+In addition to the compatibility layer the Networking Working Group has also put effort
+into building crates *directly* using the new futures API, in order to more fully
+vet that API, to provide a smoother experience for others wanting to build code using
+async/await, and to lay out a clear vision of what the new ecosystem might look like.
+
+- **[Romio][Romio-repo]**, a minimal fork of Tokio based directly on the new futures API. While
+Tokio proper aims to provide a comprehensive and opinionated story for the lowest-levels
+of async networking code, Romio covers just the essentials: an API surface very similar
+to `std::net`, but supporting async/await directly. The crate includes a good bit of
+documentation and examples, and @withoutboats has written a [blog post][romio-blog]
+detailing lessons learned through this port.
```

Comment:
Is the intention for Romio to replace Tokio? It seems unfortunate to start forking the ecosystem here if we can avoid it. Or is this just as a proof-of-concept for Tokio with futures 0.3? How does this compare to Tokio's [`async`/`await` support](https://tokio.rs/blog/2018-08-async-await/)?